### PR TITLE
add oneDNN and oneCCL samples

### PR DESCRIPTION
**Pull request by louie-tsai**
Signed-off-by: ltsai1 <louie.tsai@intel.com>
add oneDNN and oneCCL samples

### CODE REVIEW

**Comment by anjgola** [comment]
Failed due to sample.json syntax error. Please follow Instructions to move provided that have the details to validate jsons.

**Review by mkitez — APPROVED** [review]
Approving to run CI.
@louie-tsai please review failure details for the `simple_model` application. It looks like the resulting binary is not found.

File: Libraries/oneCCL/oneCCL_Getting_Started/sample.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"linux": [{
+		"env": ["source /opt/intel/oneapi/setvars.sh --ccl-configuration=cpu_gpu_dpcpp --force" ],
+		"id": "gsg",
+		"steps": [
+			"mkdir build",
+      		        "cd build",
+           		"cmake .. -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=dpcpp",
+           		"make",
+			"mpirun -n 2 ./out/sycl/sycl_allreduce_cpp_test cpu",
+			"mpirun -n 2 ./out/sycl/sycl_allreduce_cpp_test gpu"
```
Comment (mkitez) [inline Libraries/oneCCL/oneCCL_Getting_Started/sample.json:22]: I think this test should be split into 2 configurations, one for CPU the other one for GPU. Same for all other samples.

**Comment by JoeOster** [comment]
@pmpeter1 - can you please approve if appropriate

**Pull request merged**